### DISCUSSION TITLE
GitHub Actions Large Runner

### DISCUSSION THREAD

**Discussion by PHoang-AmeriCorps** [community/community#125366] [Actions]
### Select Topic Area

General

### Body

As a GitHub administrator, I would like to have option to select the region for Larger Runner due to an issue of Runner IP on same region with Azure Blob Storage recently.

If my Large Runner static IP is on same Azure region, the Firewall setting base on IP network rules will not take effect.
![image](https://github.com/community/community/assets/160646938/9a7522de-37b0-485c-9e41-3d90d97a0b9a)

https://learn.microsoft.com/en-us/azure/storage/common/storage-network-security?tabs=azure-portal#restrictions-for-ip-network-rules

The proposed solution is to create Azure VNET but I think it would be helpful if we can have option to assign region to Larger Runner.

**Comment by Jamonygr** [comment]
You are right about the Azure Storage limitation: public IP firewall rules do not help for traffic from the same Azure region in some paths, because the source may not be evaluated as the public IP you expect. For large runners that need reliable access to a locked-down storage account, Azure VNet private networking is the safer pattern. Region selection for the runner would still be useful, but for Storage firewall specifically I would design around private endpoints or VNet integration instead of relying on a same-region public static IP.
